File: cliente2/Cliente2.py
import socket

buffer = 1024
puerto = 0

class Cliente2:
    def __init__(self, ip, puerto):
        self.server = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
        self.server.connect((ip, puerto))
        print("Conectado al Servidor!")

    def enviar(self, mensaje):
        self.server.sendall(mensaje.encode('utf-8'))

    def recibir(self):
        mensaje = self.server.recv(buffer).decode('utf-8')
        return mensaje

    def cerrar_socket(self):
        self.server.close()
        print("Socket cerrado.")

def main():
    global puerto
    puerto = int(input("Introducir Puerto de conexion: "))

    cliente = Cliente2("127.0.0.1", puerto)

    while True:
        print("**************MENU**************")
        print("Elegir función:")
        print("1- Generar nombre de usuario")
        print("2- Generar contraseña")
        print("3- Cerrar Sesión")
        
        try:
            opcion = int(input())
        except ValueError:
            print("Opción no válida. Intenta de nuevo.")
            continue

        if opcion == 1:
            print("\n******Generador de nombres de usuario******\n")
            print("Escribe 'volver' para regresar al menú.")
            mensaje = input("Introducir longitud del nombre de usuario:\n")
            cliente.enviar("USERNAME:" + mensaje)

        elif opcion == 2:
            print("\n******Generador de contraseñas******\n")
            print("Escribe 'volver' para regresar al menú.")
            mensaje = input("Introducir longitud de la contraseña:\n")
            cliente.enviar("PASSWORD:" + mensaje)

        elif opcion == 3:
            cliente.enviar("Cerrar Sesion")
            cliente.cerrar_socket()
            break

        else:
            print("Opción no válida. Intenta de nuevo.")
            continue

        respuesta = cliente.recibir()
        if respuesta:
            print("Respuesta:", respuesta)
        else:
            print("\nEl mensaje no se envió, se cortó la conexión por inactividad.")
            print("Se va a cerrar el programa.")
            cliente.cerrar_socket()
            break

if __name__ == "__main__":
    main()

File: cliente2/src/Cliente2.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.io.PrintWriter;
import java.net.Socket;
import java.util.Scanner;

public class Cliente2 {
    private Socket socket;
    private BufferedReader input;
    private PrintWriter output;
    private static int puerto;

    public Cliente2() {
        try {
            System.out.println("Conectando al servidor...");
            socket = new Socket("127.0.0.1", puerto);
            input = new BufferedReader(new InputStreamReader(socket.getInputStream()));
            output = new PrintWriter(new OutputStreamWriter(socket.getOutputStream()), true);
            System.out.println("Conectado al Servidor!");
        } catch (Exception e) {
            System.out.println("No se pudo conectar al servidor.");
            e.printStackTrace();
            System.exit(0);
        }
    }

    public void enviar(String mensaje) {
        output.println(mensaje);
    }

    public String recibir() {
        try {
            return input.readLine();
        } catch (Exception e) {
            System.out.println("Error al recibir el mensaje.");
            e.printStackTrace();
            return "";
        }
    }

    public void cerrarSocket() {
        try {
            socket.close();
            System.out.println("Socket cerrado.");
        } catch (Exception e) {
            System.out.println("Error al cerrar el socket.");
            e.printStackTrace();
        }
    }

    public static int verificarUsername(String mensaje) {
        int resultado = 0;
        if (mensaje.equalsIgnoreCase("volver")) {
            resultado = 2;
        } else if (Integer.parseInt(mensaje) > 15 || Integer.parseInt(mensaje) < 5) {
            resultado = 1;
        }
        return resultado;
    }

    public static int verificarPassword(String mensaje) {
        int resultado = 0;
        if (mensaje.equalsIgnoreCase("volver")) {
            resultado = 2;
        } else if (Integer.parseInt(mensaje) >= 50 || Integer.parseInt(mensaje) < 8) {
            resultado = 1;
        }
        return resultado;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.println("Introducir Puerto de conexion");
        puerto = scanner.nextInt();

        Cliente2 cliente = new Cliente2();

        boolean loop = true;
        int menu = 0;
        String mensaje;
        int verificacion;

        while (loop) {
            switch (menu) {
                case 1:
                    System.out.println("\n******Generador de nombres de usuario******\n\n" +
                            "Escribi \"volver\" para regresar al menu.\nIntroducir longitud del nombre de usuario:");
                    mensaje = scanner.next();

                    verificacion = verificarUsername(mensaje);

                    switch (verificacion) {
                        case 1:
                            System.out.println("ERROR la cantidad de caracteres no esta entre los 5 y 15 permitidos");
                            break;

                        case 2:
                            menu = 0;
                            break;

                        default:
                            cliente.enviar("USERNAME:" + mensaje);
                            mensaje = cliente.recibir();
                            if (mensaje != null && !mensaje.isEmpty()) {
                                System.out.println("Respuesta:\n" + mensaje + "\n");
                            } else {
                                System.out.println("\nEl mensaje no se envio, se corto la conexion por inactividad\n" +
                                        "Se va a cerrar el programa\n\n");
                                cliente.cerrarSocket();
                                System.exit(0);
                            }
                            break;
                    }
                    break;

                case 2:
                    System.out.println("\n******Generador de contraseñas******\n\n" +
                            "Escribi \"volver\" para regresar al menu.\nIntroducir longitud de la contraseña:");
                    mensaje = scanner.next();

                    verificacion = verificarPassword(mensaje);

                    switch (verificacion) {
                        case 1:
                            System.out.println("ERROR la cantidad de caracteres no esta entre los 8 y 50 permitidos");
                            break;

                        case 2:
                            menu = 0;
                            break;

                        default:
                            cliente.enviar("PASSWORD:" + mensaje);
                            mensaje = cliente.recibir();
                            if (mensaje != null && !mensaje.isEmpty()) {
                                System.out.println("Respuesta:\n" + mensaje + "\n");
                            } else {
                                System.out.println("\nEl mensaje no se envio, se corto la conexion por inactividad\n" +
                                        "Se va a cerrar el programa\n\n");
                                cliente.cerrarSocket();
                                System.exit(0);
                            }
                            break;
                    }
                    break;

                case 3:
                    cliente.enviar("Cerrar Sesion");
                    cliente.cerrarSocket();
                    loop = false;
                    break;

                default:
                    System.out.println("**************MENU**************\n\n Elegir funcion\n" +
                            " 1- Generar nombre de usuario\n 2- Generar contraseña\n" +
                            " 3- Cerrar Sesion");
                    menu = scanner.nextInt();
                    break;
            }
        }

        scanner.close();
    }
}
